doc/DISTRIBUTED_PROCESSING.md

```diff
@@ -0,0 +1,91 @@
+
+## Subdivided Hydrofabric
+Certain hydrofabrics may currently require too much memory to be fully loaded by each individual MPI rank/process, which is the default behavior prior to initializing individual catchment formulations.  To work around this, it is possible to include the following flag as the final command line argument:
+
+`--subdivided-hydrofabric`
+
+This will indicate to the driver processes that the provided, full hydrofabric files should not be directly loaded.  Instead, rank/process/partition specific files should be used.  Each of these contain a subdivided, non-overlapping portion of the entire hydrofabric and are expected to correspond directly to a partition.
+
+### Driver Runtime Differences
+
+When subdivided hydrofabrics should be used, the driver processes will first check to see if the necessary subdivided hydrofabric files already exist.  If they do not, and the functionality for doing so is enabled, the driver will [generate them](#on-the-fly-generation).  If a subdivided hydrofabric is required, but files are not available and cannot be generated, the driver exits in error.
+
+### File Names
+
+The name of an expected or generated subdivided hydrofabric file is based on two things:
+* the name of the complete hydrofabric file from which is its data is obtained
+* the partition/rank id
+* 
+This will have a partition specific suffix but otherwise have the same name as the full hydrofabric files.  E.g., _catchment_data.geojson.0_ would be the subdivided hydrofabric file for _catchment_data.geojson_ specific to rank 0.  
+
+### On-the-fly Generation
+Driver processes may, under certain conditions, be able to self-subdivide a hydrofabric and generate the files when necessary.  For this to be possible, the executable must have been built with Python support (via the CMake `NGEN_ACTIVATE_PYTHON` being set to `ON`), and the [required package](DEPENDENCIES.md#the-dmodsubsetservice-package) must be installed within the Python environment available to the driver processes.
+
+## Examples
+
+### Example 1 - Full Hydrofabric
+* the CMake build directory is named `cmake-build/`
+* four MPI processes are started
+* the catchment and nexus hydrofabric files, realization config, and partition config have intuitive names and are located in the current working directory
+* all processes completely load the entire hydrofabric
+```
+mpirun -n 4 cmake-build/ngen catchment_data.geojson "" nexus_data.geojson "" realization_config.json partition_config.json
+```
+
+### Example 2 - Subdivided Hydrofabric
+* the CMake build directory is named `cmake-build/`
+* eight MPI processes are started
+* the catchment and nexus hydrofabric files, realization config, and partition config have intuitive names and are located in the current working directory
+* all processes completely load the full hydrofabric
```

Comment:
This statement doesn't seem accurate given the `--subdivided-hydrofabric` flag is used in this example.